src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -186,7 +186,7 @@ UInt SpatialPooler::getPotentialRadius() const
 
 void SpatialPooler::setPotentialRadius(UInt potentialRadius)
 {
-  potentialRadius_ = potentialRadius;
+  potentialRadius_ = std::min(potentialRadius, numInputs_); //crop to numInputs (input size)
```

Comment:
this caused the difference, because constructor does implicitly crop to bounds, while setter didn't.
TODO: do implicitly, or throw NTA_ASSERT?